orchagent/mirrororch.cpp

```diff
@@ -294,22 +300,55 @@ void MirrorOrch::deleteEntry(const string& name)
     SWSS_LOG_NOTICE("Removed mirror session %s", name.c_str());
 }
 
-bool MirrorOrch::setSessionState(const string& name, MirrorEntry& session)
+void MirrorOrch::setSessionStatus(const string& name, MirrorEntry& session)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("Setting mirroring sessions %s status\n", name.c_str());
+
+
+    setSessionState(name, session, MIRROR_SESSION_STATUS);
+}
+
+void MirrorOrch::setSessionState(const string& name, const MirrorEntry& session, const string& attr)
 {
     SWSS_LOG_ENTER();
 
     SWSS_LOG_INFO("Setting mirroring sessions %s state\n", name.c_str());
 
     vector<FieldValueTuple> fvVector;
+    string value;
```

Comment:
In this function, I think you could just check all the related information in the object `session` and save all related information that you need. I can't find a reason that you only need to store partial information based on the attribute from the input argument.

Reply:
Offline discussed. It's for efficiency purpose. Any data inconsistency should be treat as a future bug.

---
In reply to: [215803903](https://github.com/Azure/sonic-swss/pull/609#discussion_r215803903) [](ancestors = 215803903)